### DISCUSSION TITLE
We need help troubleshooting GHAS.

### DISCUSSION THREAD

**Discussion by geronethink** [community/community#55811] [Code Security]
### Select Topic Area

Product Feedback

### Body

Hi, Dear
We started testing to introduce GHAS, but we are facing continuous errors while creating the yml file and proceeding with the source code security scan. If you have the same problem as us, please suggest a solution.

>> usage .yml
`name: "CodeQL222222"

on:
  push:
    branches: [ "develop" ]
  pull_request:
    # The branches below must be a subset of the branches above
    branches: [ "develop" ]
  schedule:
    - cron: '24 23 * * 3'

jobs:
  analyze:
    name: Analyze to
    runs-on: ${{ (matrix.language == 'swift' && 'macos-latest') || 'ubuntu-latest' }}
    permissions:
      actions: write
      contents: write
      security-events: write

    strategy:
      fail-fast: false
      matrix:
        language: [ 'java' ]

    steps:
    - name: Install Depen
      run: npm install
      
    - name: Install Depen update
      run: npm update

    - name: Checkout repository
      uses: actions/checkout@v3

    - name: Initialize CodeQL
      uses: github/codeql-action/init@v2
      with:
        languages: ${{ matrix.language }}

    - name: Autobuild
      uses: github/codeql-action/autobuild@v2

    - name: Perform CodeQL Analysis
      uses: github/codeql-action/analyze@v2
      with:
        category: "/language:${{matrix.language}}"`

>> Error Log
Process completed with exit code 254.
Process completed with exit code 1.

**Comment by jketema** [comment]
Thanks for your question. I'm slightly confused by your workflow file, as on the one hand, you state that Java code should be analysed:
```
  matrix:
    language: [ 'java' ]
```
while on the other hand the restrict the languages to Swift:
```
matrix.language == 'swift'
```
What language do you want to analyse?

If you want to analyse Swift code., note that Swift support is currently in beta. To join the beta program:

1. Join the GitHub Security Lab Slack instance at [https://gh.io/securitylabslack 55](https://gh.io/securitylabslack)
2. Join the #codeql-swift-beta-lobby channel
3. Introduce yourself and your project(s) in a short message
